src/store/names.ts

```diff
@@ -0,0 +1,71 @@
+import { atom } from 'jotai';
+import { atomFamily } from 'jotai/utils';
+import { useFetch } from '@common/hooks/use-fetch';
+import { networkAtom } from './ui';
+
+interface AccountName {
+  address: string;
+  names: string[];
+}
+
+const STALE_TIME = 30 * 60 * 1000;
+
+// TODO: Should this be an `atomWithQuery`?
+async function fetchNamesByAddress(networkUrl: string, address: string): Promise<string[]> {
+  const res = await useFetch(networkUrl + `/v1/addresses/stacks/${address}`);
+  const data = await res.json();
+  return data?.names || [];
+}
+
+function makeKey(networkUrl: string, address: string): string {
+  return `${networkUrl}__${address}`;
+}
+
+function getLocalNames(networkUrl: string, address: string): [string[], number] | null {
+  const key = makeKey(networkUrl, address);
+  const value = localStorage.getItem(key);
+  if (!value) return null;
+  return JSON.parse(value);
```

Comment:
JSON.parse with invalid json throws an error. Generally best to use a wrapper lib when dealing with localStorage, which contains try/catch logic, and handling of things like `'true'` --> `true`

Reply:
I see now there is an `atomWithStorage` to use so will try to refactor.